third-party/re2/re2-src/re2/nfa.cc

```diff
@@ -700,8 +700,8 @@ bool NFA<StrPiece>::Search(const StrPiece& text, const StrPiece& const_context,
     for (int i = 0; i < nsubmatch; i++)
       submatch[i].set_ptr_end(match_[2 * i], match_[2 * i + 1]);
     if (ExtraDebug)
-      fprintf(stderr, "match (%td,%td)\n",
-              match_[0] - btext_, match_[1] - btext_);
+      fprintf(stderr, "match (%td,%ld)\n",
+              (long)(match_[0] - btext_), (long)(match_[1] - btext_));
```

Comment:
I changed it because I was getting compiler warnings. Our RE2 patches make NFA::Search a tempalet accepting StingPiece or FilePiece. When it's StringPiece, you are right that it's supposed to be ptrdiff_t. When it's FilePiece though, the subtraction is going to yield an int64_t.

I don't really care, since it's debugging code though. But do you have a suggestion for solving the problem in a better way (that gets us past the compiler error I was fixing?)